src/components/CustomAppBar.tsx

```diff
@@ -1,67 +1,86 @@
 import { AppBar, Button, Toolbar } from '@mui/material';
 import { Link as RouterLink, useNavigate } from 'react-router-dom';
 import HomeTwoToneIcon from '@mui/icons-material/HomeTwoTone';
-import React from 'react';
+import React, { useState } from 'react';
 import AccountCircleTwoToneIcon from '@mui/icons-material/AccountCircleTwoTone';
 import LogoutIcon from '@mui/icons-material/Logout';
 import { logoutUser } from '../api/auth';
+import ErrorModal from './ErrorModal';
 
 const CustomAppBar: React.FC = () => {
   const navigate = useNavigate();
+  const [isErrorModalOpen, setIsErrorModalOpen] = useState<boolean>(false);
+  const [errorDescription, setErrorDescription] = useState<string | null>(null);
 
   const handleLogout = async () => {
     try {
       const response = await logoutUser();
       if (response.ok) {
         navigate('/login');
+      } else {
+        setErrorDescription('Something went wrong, please try again!');
+        setIsErrorModalOpen(true);
       }
     } catch (error) {
-      console.error('An error occurred during logout: ', error);
+      setErrorDescription('Something went wrong, please try again!');
+      setIsErrorModalOpen(true);
     }
   };
```

Comment:
To avoid duplicate code, something like:
 ```
 const handleError = () => {
    setErrorDescription('Something went wrong, please try again!');
    setIsErrorModalOpen(true);
  };

  const handleLogout = async () => {
    try {
      const response = await logoutUser();
      if (response.ok) {
        navigate('/login');
      } else {
        handleError();
      }
    } catch (error) {
      handleError();
    }
  };
```